### ISSUE TITLE
the progress fails in error

### ISSUE THREAD

**Issue by melhossarySG** [jasonhensler/Jira-Reindex#1]
Hello,

I tried to execute this script from a jenkins job, when it hit 55% progress the the script returned a login error.

However when I check in Jira, I can see that the indexing is still progressing.

Is that something you have encountered before ? any ideas how can I fix this ?

016-05-17_20:30:01 -  Re-Indexing is at 55%
2016-05-17_20:32:44 - Error: login unsuccessful!
2016-05-17_20:32:44 -  Error retrieving progress!
2016-05-17_20:32:44 - ------------------- Http raw response ---------------------------
2016-05-17_20:32:44 - HTTP/1.1 303 See Other
Server: Apache-Coyote/1.1
X-AREQUESTID: 1232x119173x1
X-ASESSIONID: c74v4f
X-Seraph-LoginReason: AUTHENTICATED_FAILED
X-AUSERNAME: jirauser
Location: https://jiraserver/jira/rest/api/2/reindex/
Retry-After: 1
Cache-Control: no-cache, no-store, no-transform
X-Content-Type-Options: nosniff
Content-Type: application/json;charset=UTF-8
Transfer-Encoding: chunked
Date: Tue, 17 May 2016 18:32:44 GMT

{"progressUrl":"/secure/admin/jira/IndexProgress.jspa?taskId=3","currentProgress":55,"type":"BACKGROUND","submittedTime":"2016-05-17T16:09:42.163+0200","startTime":"2016-05-17T16:09:42.167+0200","success":false}
2016-05-17_20:32:44 - ------------------- ---------------------------
Build step 'Execute shell' marked build as failure
Collecting metadata...
Metadata collection done.
Finished: FAILURE

Thank you

Regards,

**Comment by melhossarySG** [comment]
Also, the re-index took in total around 6 hours. Jira's REST timeout session is by default 5 hours, do we need to increase the REST session timeout ?

Kindly please advise.

**Comment by BoB-PosT** [comment]
Yes that could very likely be part of the problem.  Can you provide some statistics on your instance to try and shine some light upon the fact that the re-index is taking a rather long time?  Also, are you using the ScriptRunner plugin?

**Comment by melhossarySG** [comment]
What kind of statistics would you need ?
and no what is ScriptRunner plugin? it is not available in the list of plugins on http://updates.jenkins-ci.org/update-center.json

the script is executed by Build execute shell from Jenkins.

**Comment by BoB-PosT** [comment]
The ScriptRunner plugin would be for JIRA, not Jenkins.
I would be interested to know how many total Issues are in your JIRA instance, as well as how many Custom Fields, Projects, and Workflows.  All of these items can be found on the System Info page under the Database Statistics section.

I am curious about whether or not ScriptRunner is installed on JIRA because in my experience it can cause a re-index to run up to 4 times longer than if it wasn't.

**Comment by melhossarySG** [comment]
Hello Chris

here are the stats

Issues  652541
Projects    999
Custom Fields   861
Workflows   1369
Attachments 142890
Comments    483556
Users   14140
Groups  60721

and yes Script Runner is installed on our Jira.

Regards,
